prov/efa/src/efa_hmem.c

```diff
@@ -142,6 +133,23 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 	else
 		info->p2p_required_by_impl = true;
 
+	info->p2p_disabled_by_user = ofi_hmem_p2p_disabled();
+
+	if (info->p2p_disabled_by_user) {
+		/* Don't need to check p2p support */
+		efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_CUDA);
+		return 0;
+	}
+
+	cuda_ret = ofi_cudaMalloc(&ptr, len);
+	if (cuda_ret != cudaSuccess) {
```

Comment:
If possible, I suggest adding unit tests to make sure:
- If user disables p2p, there is no `ibv_reg_mr` calls during domain open.
- FI_HMEM_SYSTEM always supports p2p and is never disabled.